setup.py

```diff
@@ -92,92 +103,82 @@ def run(self):
                         os.remove(compiled_path)
 
 
-class MakeStdNames(Command):
-    """
-    Generates the CF standard name module containing mappings from
-    CF standard name to associated metadata.
+def compile_pyke_rules(cmd, directory):
+    # Call out to the python executable to pre-compile the Pyke rules.
+    # Significant effort was put in to trying to get these to compile
+    # within this build process but there was no obvious way of finding
+    # a workaround to the issue presented in
+    # https://github.com/SciTools/iris/issues/2481.
 
-    """
-    description = "generate CF standard name module"
-    user_options = []
+    shelled_code = textwrap.dedent("""\
 
-    def initialize_options(self):
-        pass
+    import os
+    from pyke import knowledge_engine
 
-    def finalize_options(self):
-        pass
+    # Monkey patch the load method to avoid "ModuleNotFoundError: No module
+    # named 'iris.fileformats._pyke_rules.compiled_krb'". In this instance
+    # we simply don't want the knowledge engine, so we turn the load method
+    # into a no-op.
+    from pyke.target_pkg import target_pkg
+    target_pkg.load = lambda *args, **kwargs: None
 
-    def run(self):
-        cmd = std_name_cmd('lib')
-        self.spawn(cmd)
+    # Compile the rules by hand, without importing iris. That way we can
+    # avoid the need for all of iris' dependencies being installed.
+    os.chdir(os.path.join('{bld_dir}', 'iris', 'fileformats', '_pyke_rules'))
 
+    knowledge_engine.engine('')
```

Comment:
Compiling the rules and putting them in the right place thru `python setup.py install` isn't working from what I can see ...

Reply:
They weren't for standard_names (at least, that is what I observed in an earlier commit!)